pomegranate/distributions/distributions.pyx

```diff
@@ -299,8 +299,8 @@ cdef class Distribution(Model):
 				dist = {int(key) : value for key, value in dp.items()}
 			elif d['dtype'] == 'float':
 				dist = {float(key) : value for key, value in dp.items()}
-			elif '.' in d['dtype']:
-				dtype = d['dtype'].split('.')[-1]
+			elif d['dtype'].startswith('numpy.'):
```

Comment:
@alexhenrie You have done a lot of nice cleanups and checks.  Where might one put a check on C ordering for numpy objects?  That as been a recurring issue for repeatability.

Reply:
I'm not sure what you mean. This function only uses 1-dimensional numpy arrays, so there is no difference between C and Fortran array orderings. In what situation are you seeing an ordering mixup?

Reply:
@alexhenrie I was prompted by this PR from @MichaelMarien [1] to think about the ordering issue.  I've never dug into it deeply.  I wonder what happens if someone passes in a 1D slice from a 3D array?  I found myself in the past having to insert a `copy()` after doing a transpose such as `d.fit(s.values.T.copy())` otherwise I saw strange results as in this conversation [2] and this conversation [3].

[1] https://github.com/jmschrei/pomegranate/pull/740
[2] https://github.com/jmschrei/pomegranate/issues/414#issuecomment-390734369
[3] https://github.com/jmschrei/pomegranate/issues/400#issuecomment-375742716

Reply:
I a rewrite of my question would be "Is there a check for row-major ordering that might make sense in some place like this?"

Reply:
Array ordering can only be a problem when a multidimensional array is passed to Pomegranate by an external program and Pomegranate takes a pointer to the array data without checking the array order. I looked through the code and the only places where I believe that that can still happen are all in the k-means code. I have sent pull request https://github.com/jmschrei/pomegranate/pull/753 to address the k-means problems.

Reply:
To answer your question about what happens if a noncontiguous array slice is passed in:

```
>>> numpy.array([[1,2],[3,4]])[:,1].flags
  C_CONTIGUOUS : False
  F_CONTIGUOUS : False
  OWNDATA : False
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False

>>> numpy.array(numpy.array([[1,2],[3,4]])[:,1]).flags
  C_CONTIGUOUS : True
  F_CONTIGUOUS : True
  OWNDATA : True
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False
```

As you can see, when the noncontiguous array is passed to the numpy.array function, it is always converted to a contiguous array.

Reply:
Awesome, thanks!